### PR TITLE
Commerce 997 & 1216 Add name fields to user edit page and fix permissions allowing Account Admin update permissions

### PR DESCRIPTION
**Pull request by alecsloan**

### CODE REVIEW

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
force push reflects rebase

File: commerce-account-web/src/main/java/com/liferay/commerce/account/web/internal/portlet/action/EditCommerceAccountUserMVCActionCommand.java [hunk 1]
```diff
@@ -218,92 +228,155 @@ protected void updatePassword(
...
+			User user = _userLocalService.getUser(userId);
+
+			Date birthday = user.getBirthday();
+
+			Calendar birthdayCal = CalendarFactoryUtil.getCalendar(
+				birthday.getTime());
+
+			// Update user
+
+			_userLocalService.updateUser(
```
Comment (marco-leo) [inline commerce-account-web/src/main/java/com/liferay/commerce/account/web/internal/portlet/action/EditCommerceAccountUserMVCActionCommand.java:366]: Wrap this method in the AccountUSerService and inside that check for permisison to edit th eaccount and manage members

**Review by marco-leo — CHANGES_REQUESTED** [review]

File: commerce-account-web/src/main/java/com/liferay/commerce/account/web/internal/portlet/action/EditCommerceAccountUserMVCActionCommand.java (hunk 1 above)
Comment (alecsloan) [reply commerce-account-web/src/main/java/com/liferay/commerce/account/web/internal/portlet/action/EditCommerceAccountUserMVCActionCommand.java:366]: See the updated file plz

**Comment by alecsloan** [comment]
ci:test

File: commerce-account-web/src/main/resources/META-INF/resources/edit_user.jsp [hunk 2]
```diff
@@ -189,79 +191,103 @@ else {
 							</c:otherwise>
 						</c:choose>
 					</div>
+
+					<div class="col-lg-4">
+						<c:choose>
```
Comment (ethanbustad) [inline commerce-account-web/src/main/resources/META-INF/resources/edit_user.jsp:196]: Let's combine these `c:choose` tags, yeah? We should only have to do the test once

File: commerce-account-web/src/main/resources/META-INF/resources/edit_user.jsp [hunk 3]
```diff
@@ -38,6 +38,8 @@ else {
 		passwordResetDisabled = true;
 	}
 }
+
+boolean canEditUser = (selectedUser.getUserId() == user.getUserId()) || commerceAccountDisplayContext.hasCommerceAccountModelPermissions(commerceAccount.getCommerceAccountId(), CommerceAccountActionKeys.MANAGE_MEMBERS);
```
Comment (ethanbustad) [inline commerce-account-web/src/main/resources/META-INF/resources/edit_user.jsp:42]: This should really be a conditional. I think it's a Liferay rule to not set the result of a boolean expression to a variable.

    boolean canEditUser = false;
    if (thing1 || thing2) {
        canEditUser = true;
    }

File: commerce-account-web/src/main/resources/META-INF/resources/view.jsp [hunk 4]
```diff
@@ -26,7 +26,7 @@ List<CommerceAccount> commerceAccounts = commerceAccountDisplayContext.getCommer
 	<c:when test="<%= commerceAccountDisplayContext.getCommerceSiteType() == CommerceAccountConstants.SITE_TYPE_B2C %>">
 		<liferay-util:include page="/edit_user.jsp" servletContext="<%= application %>" />
 	</c:when>
-	<c:when test="<%= (commerceAccounts.size() > 1) || commerceAccountDisplayContext.hasManageCommerceAccountPermissions() %>">
+	<c:when test="<%= (commerceAccounts.size() > 1) && commerceAccountDisplayContext.hasManageCommerceAccountPermissions() %>">
 		<liferay-util:include page="/view_account_list.jsp" servletContext="<%= application %>" />
 	</c:when>
 	<c:when test="<%= !commerceAccounts.isEmpty() && commerceAccountDisplayContext.hasCommerceAccountModelPermissions(commerceAccounts.get(0), ActionKeys.VIEW) %>">
```
Comment (ethanbustad) [inline commerce-account-web/src/main/resources/META-INF/resources/view.jsp:32]: With the change you've applied, if a user doesn't `hasManageCommerceAccountPermissions`, I think we should potentially check `hasCommerceAccountModelPermissions` for _each_ account in the list (if it's not empty). Before your change, we knew that if we got to the 3rd `when` test then there's only 1 account in the list, but we can't assume that to be true any more... @marco-leo what do you think?

File: commerce-account-web/src/main/java/com/liferay/commerce/account/web/internal/portlet/action/EditCommerceAccountUserMVCActionCommand.java [hunk 5]
```diff
@@ -218,92 +229,157 @@ protected void updatePassword(
 			ActionResponse actionResponse)
 		throws Exception {
```
Comment (ethanbustad) [inline commerce-account-web/src/main/java/com/liferay/commerce/account/web/internal/portlet/action/EditCommerceAccountUserMVCActionCommand.java:230]: wow... this seems like a lot of logic for us to maintain.... this is the only way to expose this functionality?

File: commerce-account-web/src/main/resources/META-INF/resources/view.jsp (hunk 4 above)
Comment (marco-leo) [reply commerce-account-web/src/main/resources/META-INF/resources/view.jsp:32]: @ethanbustad  We do already this is CommerceAccountClayTable, that is the data provider for that list

File: commerce-account-web/src/main/java/com/liferay/commerce/account/web/internal/portlet/action/EditCommerceAccountUserMVCActionCommand.java (hunk 5 above)
Comment (marco-leo) [reply commerce-account-web/src/main/java/com/liferay/commerce/account/web/internal/portlet/action/EditCommerceAccountUserMVCActionCommand.java:230]: I agree, what do you suggest to simplify?

File: commerce-account-web/src/main/resources/META-INF/resources/view.jsp (hunk 4 above)
Comment (ethanbustad) [reply commerce-account-web/src/main/resources/META-INF/resources/view.jsp:32]: it just seems weird that we check `hasCommerceAccountModelPermissions` for the first element in the list before rendering `view_account.jsp` -- we don't know there's only one element in the list, so why single out the first element for a permission check?

Comment (marco-leo) [reply commerce-account-web/src/main/resources/META-INF/resources/view.jsp:32]: @alecsloan Thanks to this commerce, I just realized that the logic is (commerceAccounts.size() > 1) || commerceAccountDisplayContext.hasManageCommerceAccountPermissions() not &&

Comment (alecsloan) [reply commerce-account-web/src/main/resources/META-INF/resources/view.jsp:32]: So what do I need to do here? Change it back to or? I don't think that logic was correct.

File: commerce-account-web/src/main/resources/META-INF/resources/edit_user.jsp (hunk 2 above)
Comment (alecsloan) [reply commerce-account-web/src/main/resources/META-INF/resources/edit_user.jsp:196]: Wow why did it do that, Of course!

**Pull request merged**